src/handlers/generate-kubeconfig.handler.ts

```diff
@@ -0,0 +1,116 @@
+import util from 'util';
+
+import { Logger } from '../logger.service/logger';
+import { ConfigService } from '../config.service/config.service';
+
+const path = require('path');
+const fs = require('fs');
+const pem = require('pem');
+
+export async function generateKubeconfigHandler(
+    argv: any,
+    configService: ConfigService,
+    logger: Logger
+) {
+    // Check if we already have generated a cert/key
+    let kubeConfig = configService.getKubeConfig();
+
+    if (kubeConfig == undefined) {
+        logger.info('No KubeConfig has been generated before, generating key and cert for local daemon...');
+
+        // Create and save key/cert
+        const createCertPromise = new Promise<void>(async (resolve, reject) => {
+            pem.createCertificate({ days: 999, selfSigned: true }, async function (err: any, keys: any) {
+                if (err) {
+                    throw err;
+                }
+
+                // Get the path of where we want to save
+                const pathToConfig = path.dirname(configService.configPath());
+                const pathToKey = `${pathToConfig}/kubeKey.pem`;
+                const pathToCert = `${pathToConfig}/kubeCert.pem`;
+
+                // Now save the key and cert
+                await fs.writeFile(pathToKey, keys.serviceKey, function (err: any) {
+                    if (err) {
+                        logger.error('Error writing key to file!');
+                        reject();
+                        return;
+                    }
+                    logger.debug('Generated and saved key file');
+                });
+                await fs.writeFile(pathToCert, keys.certificate, function (err: any) {
+                    if (err) {
+                        logger.error('Error writing cert to file!');
+                        reject();
+                        return;
+                    }
+                    logger.debug('Generated and saved cert file');
+                });
+
+                // Generate a token that can be used for auth
+                const randtoken = require('rand-token');
+                const token = randtoken.generate(128) + '++++';
+
+                // Find an open port, define it here as if the config has already been created, this codeblock will never be executed
+                const findPort = require('find-open-port');
+                const localPort = new Promise<number>(async (resolve, _) => {
+                    findPort().then((port: any) => {
+                        resolve(port);
+                    });
+                });
+
```

Comment:
Can't this be simplified to?:

```suggestion
                const localPort = await findPort();

```

Because it looks like `findPort` returns a promise.